src/query_graph/path_tree.rs

```diff
@@ -111,13 +111,59 @@ impl OpPathTree {
         }
         Ok(true)
     }
+
+    fn fmt_internal(
+        &self,
+        f: &mut Formatter<'_>,
+        indent: &String,
```

Comment:
```suggestion
        indent: &str,
```
Should take a `&str` instead of an owned String whenever possible, it's more flexible and `String` easily dereferences to `&str`.